pkg/api-server/service_insight_endpoints.go

```diff
@@ -71,8 +72,11 @@ func (s *serviceInsightEndpoints) listResources(request *restful.Request, respon
 		return
 	}
 
-	restList := s.expandInsights(serviceInsightList)
-	restList.Total = uint32(len(restList.Items))
+	items := s.expandInsights(serviceInsightList)
+	restList := rest.ResourceList{
+		Total: uint32(len(items)),
+		Items: items,
```

Comment:
Shouldn't we just set `json:"omitempty"` on all ResourceLists?

Reply:
I don't think so it's a little weird to have this field omitted from a list response and get: `{"total": 0, "next": null}`

Reply:
Right if we set `items []SomeResource \`json:"Items,omitempty"\`` we wouldn't ever get `items: null`, would we? Or maybe I'm missing something

Reply:
Ah you're saying it should never be omitted. Ok, i guess that's also a valid API guarantee. Arg golang